lib/standard/stream.nit

```diff
@@ -49,18 +50,19 @@ abstract class Reader
 	fun read_byte: nullable Int is abstract
 
 	# Read at most i bytes
-	fun read(i: Int): String
+	fun read(i: Int): Bytes
```

Comment:
Agreed, that was a bit of an inner struggle for me too.

I still think that since we're dealing with a purely byte-oriented entity, the base operations (i.e. write, read, read_all, etc.) should work with bytes.

However we discussed with @xymus about all this, and it seems that from his point of view, these operations should continue working on String, since most developers will write pure text.

I can rename most byte-oriented services then, as with the present scheme for read_all and read_bytes (which will be renamed in this case, probably to read_all_bytes).

Another solution is to have all base names related to bytes and have string equivalents with _string appended to the base name, your choice !